husarion_ugv_navigation/src/map_autosaver_node.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "husarion_ugv_navigation/map_autosaver_node.hpp"
+
+namespace husarion_ugv_navigation {
+
+constexpr auto SAVE_MAP_CONNECTION_TIMEOUT = 2s;
+constexpr auto MIN_SAVE_MAP_PERIOD = 5s;
+
+AutosaveMapNode::AutosaveMapNode(const std::string &node_name,
+                                 const rclcpp::NodeOptions &options)
+    : Node(node_name, options), save_map_period_(MIN_SAVE_MAP_PERIOD) {
+  this->declare_parameter<double>("autosave_period",
+                                  MIN_SAVE_MAP_PERIOD.count());
+
+  this->declare_parameter<std::string>("map_directory", "/maps/map");
+  double period = MIN_SAVE_MAP_PERIOD.count();
```

Comment:
`MIN_SAVE_MAP_PERIOD.count();` seems to be unnecessary because it is overwrite. 
```suggestion
  double period;
```